tests/functional/aws-node-sdk/test/multipleBackend/put.js

```diff
@@ -113,6 +141,147 @@ describe('MultipleBackend put object', () => {
+                                `got error ${err}`);
+                            assert.strictEqual(res.ETag, `"${correctMD5}"`);
+                            done();
+                        });
+                    });
+                });
+            });
+
+            itSkipIfNotMultiple('should put a large object to AWS', done => {
+                const params = { Bucket: bucket, Key: key,
+                    Body: bigBody,
+                    Metadata: { 'scal-location-constraint': 'aws-test' } };
+                s3.putObject(params, err => {
+                    assert.equal(err, null, 'Expected sucess, ' +
+                        `got error ${err}`);
+                    s3.getObject({ Bucket: bucket, Key: key }, (err, res) => {
+                        assert.strictEqual(err, null, 'Expected success, ' +
+                            `got error ${err}`);
+                        assert.strictEqual(res.ETag, `"${bigMD5}"`);
+                        awsS3.getObject({ Bucket: awsBucket, Key: key },
+                        (err, res) => {
+                            assert.strictEqual(err, null, 'Expected success, ' +
+                                `got error ${err}`);
+                            assert.strictEqual(res.ETag, `"${bigMD5}"`);
+                            done();
+                        });
+                    });
+                });
+            });
+
+            itSkipIfNotMultiple('should put objects with same key to AWS ' +
+            'then file, and object should only be present in file', done => {
+                const params = { Bucket: bucket, Key: key,
+                    Body: body,
+                    Metadata: { 'scal-location-constraint': 'aws-test' } };
+                s3.putObject(params, err => {
+                    assert.equal(err, null, 'Expected success, ' +
+                        `got error ${err}`);
+                    params.Metadata = { 'scal-location-constraint': 'file' };
+                    s3.putObject(params, err => {
+                        assert.equal(err, null, 'Expected success, ' +
+                            `got error ${err}`);
+                        s3.getObject({ Bucket: bucket, Key: key },
+                        (err, res) => {
+                            assert.equal(err, null, 'Expected success, ' +
+                                `got error ${err}`);
+                            assert.strictEqual(
+                                res.Metadata['scal-location-constraint'],
+                                'file');
+                            awsS3.getObject({ Bucket: awsBucket,
+                            Key: key }, err => {
+                                assert.strictEqual(err.code, 'NoSuchKey');
+                                done();
+                            });
+                        });
+                    });
+                });
+            });
+
+            itSkipIfNotMultiple('should put objects with same key to file ' +
```

Comment:
can you add a test with maybe file first and then mem where you check to make sure the ds array doesn't have the mem stuff anymore?